src/algebra/lie/basic.lean

```diff
@@ -846,6 +934,23 @@ end quotient
 
 end lie_submodule
 
+section lie_algebra_properties
+
+variables (R : Type u) (L : Type v) (M : Type w)
+variables [comm_ring R] [lie_ring L] [lie_algebra R L] [add_comm_group M] [module R M]
+variables [lie_ring_module L M] [lie_module R L M]
+
+/-- A Lie module is irreducible if it is zero or its only non-trivial Lie submodule is itself. -/
+class lie_module.is_irreducible : Prop :=
+(irreducible : ∀ (N : lie_submodule R L M), N ≠ ⊥ → N = ⊤)
+
+/-- A Lie algebra is simple if it is irreducible as a Lie module over itself via the adjoint
+action, and it is non-Abelian. -/
+class lie_algebra.is_simple : Prop :=
+(simple : lie_module.is_irreducible R L L ∧ ¬is_lie_abelian L)
```

Comment:
Using another typeclass as a field seems strange to me - wouldn't it be better to extend `isirreducible`?

Reply:
Agreed, thanks.

I was actually just moving these definitions (because I could now state `is_irreducible` in terms of the lattice structure) but in the time since I created these I have since refactored `lie_module`s so that I can now set things up as you suggest. Nice!